### PR TITLE
vector_of_shared_ptr

### PR DESCRIPTION
**Pull request by Martyna-Tu**

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: vector-of-shared-ptrs
🏅 Granted 5 XP for @Martyna-Tu

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    for(int i = 0; i < count; i++){
+        std::shared_ptr<int> ptr = std::make_shared<int>(i);
+        vec.push_back(ptr);
+    }
+
+    return vec;
+}
+
+void print(std::vector<std::shared_ptr<int>> vec){
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:16]: Pass by const refrence. Try always avoid copy.

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    for(const auto& el : vec){
+        std::cout << *el << " ";
+    }
+    std::cout << "\n";
+}
+   
+void add10(std::vector<std::shared_ptr<int>> vec){
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:24]: Pass by const refrence. Try always avoid copy.

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    
+    for(auto& el : vec){
+        if(el){ 
+            *el += 10;    
+       }
+    }
+}   
+
+void sub10(int* const ptr){
+    if(ptr){
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:34]: Add space after `if`, `for`, `while` `function` etc...

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        std::shared_ptr<int> ptr = std::make_shared<int>(i);
+        vec.push_back(ptr);
+    }
+
+    return vec;
+}
+
+void print(std::vector<std::shared_ptr<int>> vec){
+
+    for(const auto& el : vec){
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:18]: Add space after `if`, `for`, `while` `function` etc...

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        std::cout << *el << " ";
+    }
+    std::cout << "\n";
+}
+   
+void add10(std::vector<std::shared_ptr<int>> vec){
+    
+    for(auto& el : vec){
+        if(el){
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:27]: Add space after `if`, `for`, `while` `function` etc...

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**